src/Fable.React/Fable.Helpers.React.fs

```diff
@@ -903,45 +905,82 @@ module ReactElementType =
     let inline create<'props> (comp: ReactElementType<'props>) (props: 'props) (children: ReactElement seq): ReactElement =
         createElement(comp, props, children)
 
-type PropsEqualityComparison<'props> = 'props -> 'props -> bool
-
-[<Import("memo", from="react")>]
-let private reactMemo<'props> (render: 'props -> ReactElement) : ReactComponentType<'props> =
-    jsNative
-
-/// React.memo is a higher order component. It’s similar to React.PureComponent but for function components instead of
-/// classes.
+    /// React.memo is a higher order component. It’s similar to React.PureComponent but for function components instead of
+    /// classes.
+    ///
+    /// If your function component renders the same result given the same props, you can wrap it in a call to React.memo
+    /// for a performance boost in some cases by memoizing the result. This means that React will skip rendering the
+    /// component, and reuse the last rendered result.
+    ///
+    /// By default it will only shallowly compare complex objects in the props object. If you want control over the
+    /// comparison, you can use `memoWith`.
+    [<Import("memo", from="react")>]
+    let memo<'props> (render: 'props -> ReactElement) : ReactComponentType<'props> =
+        jsNative
+
+    [<Import("memo", from="react")>]
+    let private reactMemoWith<'props> (render: 'props -> ReactElement, areEqual: PropsEqualityComparison<'props>) : ReactComponentType<'props> =
+        jsNative
+
+    /// React.memo is a higher order component. It’s similar to React.PureComponent but for function components instead of
+    /// classes.
+    ///
+    /// If your function component renders the same result given the same props, you can wrap it in a call to React.memo
+    /// for a performance boost in some cases by memoizing the result. This means that React will skip rendering the
+    /// component, and reuse the last rendered result.
+    ///
+    /// This version allow you to control the comparison used instead of the default shallow one by provide a custom
+    /// comparison function.
+    let memoWith<'props> (areEqual: PropsEqualityComparison<'props>) (render: 'props -> ReactElement) : ReactComponentType<'props> =
+        reactMemoWith(render, areEqual)
```

Comment:
The F# compiler (and thus Fable) translates module functions with tuple and curried arguments in the same way, so you should be able to add the `Import` attribute directly here.

Reply:
Yes but I can't reverse the arguments :)

Reply:
Ah, didn't notice that, sorry! 😅 Then this works :+1: There's still a way to use only attributes, but it's a bit hacky and I haven't used it very much:

```fsharp
[<Import("memo", from="react"); Emit("$0($2, $1)")>]
let memoWith<'props> (areEqual: PropsEqualityComparison<'props>) (render: 'props -> ReactElement) : ReactComponentType<'props> = jsNative
```

Reply:
I love Emit but it might be a little bit too hacky :)